### PR TITLE
ENH: link stable sitemap

### PR DESCRIPTION
**Pull request by jklymak**
Much pared-down version of #47.  

Instead of redirecting the whole root tree, it now simply takes root out of the canonical search path in `sitemap.xml` and `robots.txt`, and instead suggests going to the softlinked `stable->3.3.3`.  `sitemap.xml` and `robots.txt` would need to be manually updated at release (though robots.txt should be fine until Matplotlib 6).

## Todo:

- [ ] canonical links will still need to point to `matplotlib.org/stable/`

### CODE REVIEW

**Comment by jklymak** [comment]
Someone who understands `sitemap.xml` and `robots.txt` better than me should definitely jump in here if they don't think this will work....

**Comment by jklymak** [comment]
BTW do we want /stable/ or /latest/?  I guess I like `/stable/` as a parallel to `/dev/` however, I probably typed `/latest/` 50% of the time working on this...

**Comment by dstansby** [comment]
> BTW do we want /stable/ or /latest/? I guess I like `/stable/` as a parallel to `/dev/` however, I probably typed `/latest/` 50% of the time working on this...

Definitely `stable`, by default on readthedocs (which many many many python projects use) `stable` means the latest release, and `latest` means the latest commit on the main branch.

p.s. I'm half following all of this, thanks a lot for the PRs!

File: sitemap.xml [hunk 1]
```diff
@@ -0,0 +1,195 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+   <url>
+      <loc>http://matplotlib.org/stable/</loc>
```
Comment (jklymak) [inline sitemap.xml:5]: Note these should really have a `<last-modified>` tag, but I think they will work without.

**Comment by jklymak** [comment]
A way forward to redirect root level instead of replacing root level by soft links to `stable` (#47) would be to figure out the newest version (usual `stable`, but other versions if the file is now removed) and replace all the root level html files with a refresh:  

```html
<!DOCTYPE HTML>                                                                 
<html lang="en">                                                                
    <head>                                                                      
        <meta charset="utf-8">
        <meta http-equiv="refresh" content="0;url={{NEWEST_VERSION}}/PATH" />       
        <link rel="canonical" href="{{NEWEST_VERSION}}/PATH" />                     
    </head>                                                                                                                                                                   
    <body>                                                                      
        <h1>                                                                    
            The page been moved to <a href="{{NEWEST_VERSION}}/PATH"</a>
        </h1>                                                                   
    </body>                                                                     
</html>
```

Yet another option is change the hosting from GitHub pages, and be able to include our own `.htaccess`

**Comment by QuLogic** [comment]
This seems pretty reasonable, but all I did was read the docs that you linked, so I don't now about the SEO implications.

You may have missed a few project subdirectories:
* `/matplotblog` definitely should be crawled
* `/pytest-mpl` has a `gh-pages` branch, but seems to be broken
* `/basemap` maybe should still be crawled
* `/cycler` should be crawled
* `/mpl-altair` might be a bit dead, but can still be crawled
* `/mpl-bench` needs to be automated, but can be crawled
* `/devdocs` could be crawled, but should be marked as updated daily at least

I have some ideas of how to proceed on the redirects now, but no time to try them just yet. Maybe after the 3.4.0 release candidate.

**Comment by jklymak** [comment]
Great  I can add those.  But how do you know those are all there?  I can see how the main docs are built, but how do the other ones get injected?  Do they just have GitHub pages that deploy onto ours?

**Comment by QuLogic** [comment]
Any repo at `github.com/USER/REPO` deploys to `USER.github.io/REPO` when it has a `gh-pages` branch. We just also have a CNAME setup to use `matplotlib.org` instead of `USER.github.io`. I'm not actually sure what happens if the main site has a page called `REPO` that would shadow the repo-specific one.

**Pull request merged**